e2eAIOK/common/trainer/data/data_builder_cv.py

```diff
@@ -21,20 +21,26 @@ def get_dataloader(self):
         pin_memory = self.cfg.pin_mem if "pin_mem" in self.cfg else False
 
         ### check whether distributed or enable_ipex
+        ## For train
         if ext_dist.my_size > 1:
             num_tasks = ext_dist.dist.get_world_size()
             global_rank = ext_dist.dist.get_rank()
             sampler_train = torch.utils.data.DistributedSampler(
                 self.dataset_train, num_replicas=num_tasks, rank=global_rank, shuffle=True, drop_last= drop_last)
+        else:
+            sampler_train = torch.utils.data.RandomSampler(self.dataset_train)
+
+        ## For evaluation
+        if ext_dist.my_size > 1 and not ("ddp_eval_nosplit" in self.cfg and self.cfg.ddp_eval_nosplit): # ddp_eval_nosplit flag means do not split evaluation dataset in distribution mode
```

Comment:
How to decide when we should use or not use 'ddp_eval_nosplit'

Reply:
Depends on your use case. 
If evaluation tooks too long time and the metric is simple, can split it and then merge.
But some complex metric is not proper to be merged, such as if they depends on the whole data. Or evaluation is very quick, we can calculate once to skip spliting and merging.

Reply:
I got that, my concern is that 'ddp_eval_nosplit' is an advanced configuration and required user already did recursive tests and they are finding current result is not bad. Which will be a bad user experience.
My suggestion here is for 'known' dataset, can we give default configuration on 'dpp_eval_nosplit' and only when this is 'unknown'/'new' dataset,  'ddp_eval_nosplit' is required user's configuration.

Reply:
We have default option for "known" dataset, all DENAS related cases will split evaluation data; all Model Adapter related cases will not split, this is defined in "e2eAIOK/ModelAdapter/default_ma.conf".